### DISCUSSION TITLE
Adjust Lighthouse score for nested iframes

### DISCUSSION THREAD

**Discussion by BrunoLeSquirrel** [GoogleChrome/lighthouse#16077] [Q&A]
I want to measure the lighthouse score for a collection of webpages which all feature nested iframes.
Let's imagine it's a stack like A-B-C-some_times_even_D.
Whilst I can remove A I cannot remove B. But I want to measure the times for the various C (and ...D)
Whilst I am interested in how A or B performs, it is really C and D whose performance interests me.
Is it possible to to run lighthouse on the webpages and adjust the input so that contributions from A and B are ignored?
For example, I know the timeOrigin of all A, B, C 
Can I tell lighthouse to ignore anything before timeOrigin=C
Is there any such hack supported?

**Comment by adamraine** [comment]
Lighthouse can only audit the main frame. It looks at iframes to see how they affect the experience of the page as a whole but it is not possible to run an audit on an iframe is if it was it's own page.

> Let's imagine it's a stack like A-B-C-some_times_even_D. Whilst I can remove A I cannot remove B. But I want to measure the times for the various C (and ...D)

I'm not sure what you mean by this. Are A/B/C all frames? If so why do you say that you can remove A?

**Reply by BrunoLeSquirrel** [reply]
Hi Adam,

The iframe tree structure could be either 3 or 4 frames deep. You have some container layers (A, B) and some application layers (C, D)
Obviously you can pass the whole url into lighthouse and get back a answer.
The problem is this answers the question for what is the performance of A-B-C-D, rather than what is the performance of C-D
If I understand you correctly, there's no way to 'correct'(alter) the answer, even if I know the timestamps of all the data which should be ignored.
